### DISCUSSION TITLE
Warning about hoptodesk.com

### DISCUSSION THREAD

**Discussion by 0x100001** [rustdesk/rustdesk#2778] [General]
Someone on Discord posted about it and I didn't know where else to share my analysis results.
Maybe interesting for @zeroheure and @rustdesk too.

I had a quick look at the whole thing. The company behind it is allegedly* based in the UK (https://find-and-update.company-information.service.gov.uk/company/12249761).
The nameservers are run by Cloudflare, which is an absolute data protection disaster:
clint.ns.cloudflare.com
marge.ns.cloudflare.com

The API is at least, let's call it "interesting":
https://api.hoptodesk.com/

One of the associated servers has a cPanel installation running:
http://207.148.119.53/

When accessing the admin port of the cPanel installation you will be redirected:
https://thong.kaiidea.com:2087/

Hoptodesk seems to be connected to this in some way, which doesn't seem very trustworthy.
The registrar is again Namecheap (which is ok, I like Namecheap), but also with Whois Protection and again via Cloudflare.

The login address for cPanel seems to be a gmail: 
o(hidden)s@gmail.com 

Let's take a quick detour. The operator hosts not very trustworthy projects.
```
moneyapp.kaiidea.com 172.67.219.96 CloudFlare is on
connection.kaiidea.com 172.67.219.96 CloudFlare is on
monkeypics.kaiidea.com 172.67.219.96 CloudFlare is on
bal.kaiidea.com 104.21.43.47 CloudFlare is on
thunderad.kaiidea.com 172.67.219.96 CloudFlare is on
trueview.kaiidea.com 172.67.219.96 CloudFlare is on
thong.kaiidea.com 172.67.219.96 CloudFlare is on
ipfs.kaiidea.com 172.67.219.96 CloudFlare is on
honeykb.kaiidea.com 104.21.43.47 CloudFlare is on
smart.kaiidea.com 172.67.219.96 CloudFlare is on
monkey.kaiidea.com 172.67.219.96 CloudFlare is on
gold.kaiidea.com 104.21.43.47 CloudFlare is on
kaiidea.com 172.67.219.96 CloudFlare is on
www.kaiidea.com 172.67.219.96 CloudFlare is on
freelancer.kaiidea.com 104.21.43.47 CloudFlare is on
cloudsmartbit.kaiidea.com none CloudFlare is off
fitproperty.kaiidea.com none CloudFlare is off
deeload.kaiidea.com none CloudFlare is off
www.monkeypics.kaiidea.com none CloudFlare is off
www.fitproperty.kaiidea.com none CloudFlare is off
www.freelancer.kaiidea.com none CloudFlare is off
www.ipfs.kaiidea.com none CloudFlare is off
www.honeykb.kaiidea.com none CloudFlare is off
www.smart.kaiidea.com none CloudFlare is off
www.monkey.kaiidea.com none CloudFlare is off
www.deeload.kaiidea.com none CloudFlare is off
www.gold.kaiidea.com none CloudFlare is off
sexyfranz.com.kaiidea.com none CloudFlare is off
www.sexyfranz.com.kaiidea.com none CloudFlare is off
cpanel.kaiidea.com none CloudFlare is off
cpcalendars.kaiidea.com none CloudFlare is off
cpcontacts.kaiidea.com none CloudFlare is off
mail.kaiidea.com none CloudFlare is off
webdisk.kaiidea.com none CloudFlare is off
webmail.kaiidea.com none CloudFlare is off
in.kaiidea.com none CloudFlare is off
wwwe.kaiidea.com none CloudFlare is off
hellocheckinn.kaiidea.com none CloudFlare is off
www.hellocheckinn.kaiidea.com none CloudFlare is off
www.cloudsmartbit.kaiidea.com none CloudFlare is off
www.bal.kaiidea.com none CloudFlare is off
www.moneyapp.kaiidea.com none CloudFlare is off
freelance.kaiidea.com none CloudFlare is off
www.freelance.kaiidea.com none CloudFlare is off
www.thong.kaiidea.com none CloudFlare is off
checkinn.kaiidea.com none CloudFlare is off
www.checkinn.kaiidea.com
```

Their MySQL server is also publicly accessible, which is not good.

I have no motivation to check every single domain, but I think the overview is enough. There are some weird crypto projects behind.

Their Windows version contacts at least the following servers:
vps-08ae2688.vps.ovh.net:3478 
107.152.35.199 (Cloudfanatic.net)
174.138.26.224 (Singapore Digitalocean Llc)
207.148.119.53.vultrusercontent.com:3478

I took a closer look at the Windows client. I could not detect any malicious actions, neither during static nor dynamic analysis. However, I have to say that I did not connect to any other system, so it may be that more is requested if needed. 
Regarding the usage, the Windows service is missing, so unattended access does not work after reboot. RustDesk is much more advanced.

In conclusion, I would not recommend you to use it. The whole construct behind it is absolutely opaque. 
I recommend everyone not to use it. Often this is exactly how the classic incident response start in some companies.
The person behind that project either has no idea what he is doing, or ...

**Comment by kolAflash** [comment]
Related: #1590 _about the HopToDesk fork_

**Comment by lrq3000** [comment]
I'm just a long-time user of RustDesk and I made suggestions early on about licensing, so the following is just my opinion.

Unfortunately I don't think it's the responsibility of @rustdesk or anyone here to do anything about it, forks of open-source projects can potentially be malicious, it's a common issue but this comes with the freedom of open-sourced code.

At least you documented the issue (thank you very much @0x100001 ), so at least other prospective potential users can find this information. If someone here consider the evidence sufficient to label them as malicious, they could be reported to the authorities for cybercrime in the country where they are registered, but I doubt there will be anything done unless there is serious evidence that there is a tangible damage (ie, it would be more effective if someone who was one of their user reported a damage they suppose on their own stuff).
